### PR TITLE
Adds a reboot flag to `reset`

### PR DESCRIPTION
**Pull request by crdant**
TL;DR
-----

Simplifies resetting a node iwth a reboot flag

Details
-------

Encourages better behavior by making it easier to reboot a node after resetting it. Instead of having to run two commands to assure a clean reset:

```
$ sudo ./embedded-cluster reset
$ sudo reboot
```

you just have to

```
$ sudo ./embedded-cluster reset --reboot
```

to do the same.

My implementation shells out to the `reboot` command which takes a little longer than making the minimum required syscalls (`sync(2)` and `reboot(2)`) because of the niceties it brings along like sending a wall message and signaling processes more gracefully.

My first implementation just made the syscalls and it was a lot faster but I decided it probably makes more sense to play nice. I could be convinced it's not needed since the node shouldn't be running any other workloads and the app and cluster are already nicely taken care of.

### CODE REVIEW

**Review by laverya — APPROVED** [review]

File: cmd/embedded-cluster/uninstall.go [hunk 1]
```diff
@@ -404,6 +409,16 @@ var resetCommand = &cli.Command{
...
 
+		if c.Bool("reboot") {
+			// this is a bit more graceful that using sync(2) and reboot(2) but
+			// can take a bit longer
+			if _, err := exec.Command("reboot").Output(); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("Aborting reboot")
+		}
```
Comment (laverya) [inline cmd/embedded-cluster/uninstall.go:420]:
```suggestion
		}
```

Comment (laverya) [reply cmd/embedded-cluster/uninstall.go:420]: as it is right now, this exits with an error unless you pass in the "--reboot" flag

**Review by laverya — CHANGES_REQUESTED** [review]
Please don't make the command exit nonzero when not passing the reboot flag

**Comment by crdant** [comment]
Oops. Moved the else to the wrong place e when I switched implementations.

**Comment by ajp-io** [comment]
After reset, there's a message that says, "Node has been reset. Please reboot to ensure transient configuration is also reset." Let's make sure we don't show the reboot part if `--reboot` was passed.

File: cmd/embedded-cluster/uninstall.go (hunk 1 above)
Comment (crdant) [reply cmd/embedded-cluster/uninstall.go]: Definitely wasn't the plan. I thought I had put the `else` on the wrong `if`, which essentially I had, but it was actually that I took out a prompt for "Continue with reboot" that had been in there. Should be fixed up now.

**Comment by crdant** [comment]
> After reset, there's a message that says, "Node has been reset. Please reboot to ensure transient configuration is also reset." Let's make sure we don't show the reboot part if `--reboot` was passed.

I moved the info logging to account for this. Not sure if it was the best approach since it took the logging out of the function that does the reset.

File: cmd/embedded-cluster/uninstall.go [hunk 2]
```diff
@@ -404,6 +412,18 @@ var resetCommand = &cli.Command{
 			}
 		}
 
+		if c.Bool("reboot") {
+			if prompts.New().Confirm("Continue with reboot?", false) {
```
Comment (crdant) [inline cmd/embedded-cluster/uninstall.go:416]: I'm not sure we really need this, but it felt like the responsible thing to do.

Comment (laverya) [reply cmd/embedded-cluster/uninstall.go:416]: if you want to add a confirm here, you'll need to only do so when we don't pass "--no-prompts" IMO

Comment (ajp-io) [reply cmd/embedded-cluster/uninstall.go:416]: Given that someone passed the reboot flag, I don't think we need a confirmation for this.

**Review by laverya — APPROVED** [review]
